pandas/core/strings.py

```diff
@@ -1096,7 +1089,9 @@ def _wrap_result_expand(self, result, expand=False):
 
             if expand:
                 result = list(result)
-            return Index(result, name=name)
+                return MultiIndex.from_tuples(result, names=name)
```

Comment:
Yes. Updated #10008 for current status. The current impl can return ``MultiIndex`` for ``partition (#9773) already, but it doesn't work for ``split`` (thus changed).